examples/user-data-store/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! An example of a user data store that with an async interface that's driven by the executor on
+//! the foreign side of the FFI.  This is achieved by awaiting async callback interface methods.
+//! See the callbacks section of the manual for the technical details.
```

Comment:
I expect a bunch of compiler errors if you tried to build this.  The code will definitely need to change, but I think the general spirit is correct.